Bonobo.Git.Server/Controllers/GitController.cs

```diff
@@ -22,10 +22,13 @@ public class GitController : Controller
 
         public ActionResult SecureGetInfoRefs(String project, String service)
         {
-            if (RepositoryPermissionService.AllowsAnonymous(project)
-                || RepositoryPermissionService.HasPermission(User.Username(), project)
-                    || (String.Equals("git-upload-pack", service, StringComparison.OrdinalIgnoreCase)
-                        || UserConfiguration.Current.AllowAnonymousPush))
+            bool allowAnonClone = RepositoryPermissionService.AllowsAnonymous(project);
+            bool hasPermission = RepositoryPermissionService.HasPermission(User.Id(), project);
```

Comment:
There is no Integer overload of `hasPermission()`. Maybe you want to stick to `User.Username()`.